receiver/mongodbatlasreceiver/mongodb_event_to_logdata.go

```diff
@@ -0,0 +1,167 @@
+
+	// Attributes related to the object causing the event.
+	resourceAttrs.InsertString("org", r.Org.Name)
+	resourceAttrs.InsertString("project", r.Project.Name)
+	resourceAttrs.InsertString("cluster", r.Cluster.Name)
+	resourceAttrs.InsertString("hostname", r.Hostname)
+
+	data, err := json.Marshal(e)
+	if err != nil {
+		logger.Warn("failed to marshal", zap.Error(err))
+	}
+
+	t, err := time.Parse(layout, e.Timestamp.Date)
+	if err != nil {
+		logger.Warn("Time failed to parse correctly", zap.Error(err))
+	}
+	lr.SetTimestamp(pcommon.NewTimestampFromTime(t))
+	lr.SetObservedTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+
+	// The Message field contains description about the event,
+	// which is best suited for the "Body" of the LogRecordSlice.
+	lr.Body().SetStringVal(string(data))
+
+	// Set the "SeverityNumber" and "SeverityText" if a known type of
+	// severity is found.
+	lr.SetSeverityNumber(plog.SeverityNumberINFO)
+	lr.SetSeverityText("INFO")
+
+	attrs := lr.Attributes()
+	attrs.EnsureCapacity(totalLogAttributes)
+
+	if e.AuthType != "" {
+		attrs.InsertString("authtype", e.AuthType)
+	}
+
+	attrs.InsertString("local.ip", e.Local.IP)
+	attrs.InsertInt("local.port", int64(e.Local.Port))
+	attrs.InsertString("remote.ip", e.Remote.IP)
+	attrs.InsertInt("remote.port", int64(e.Remote.Port))
+	attrs.InsertString("uuid.binary", e.ID.Binary)
+	attrs.InsertString("uuid.type", e.ID.Type)
+	attrs.InsertInt("remote.port", int64(e.Result))
+	attrs.InsertString("log_name", r.LogName)
+
+	if e.Param.User != "" {
+		attrs.InsertString("param.user", e.Param.User)
+		attrs.InsertString("param.database", e.Param.Database)
+		attrs.InsertString("param.mechanism", e.Param.Mechanism)
+	}
+
+	return ld
+}
+
+// mongoEventToLogRecord converts model.LogEntry event to plog.LogRecordSlice and adds the resource attributes.
+func mongodbEventToLogData(logger *zap.Logger, e *model.LogEntry, r resourceInfo) plog.Logs {
+	ld := plog.NewLogs()
+	rl := ld.ResourceLogs().AppendEmpty()
+	sl := rl.ScopeLogs().AppendEmpty()
+	lr := sl.LogRecords().AppendEmpty()
+
+	resourceAttrs := rl.Resource().Attributes()
+	resourceAttrs.EnsureCapacity(totalResourceAttributes)
+
+	// Attributes related to the object causing the event.
+	resourceAttrs.InsertString("org", r.Org.Name)
+	resourceAttrs.InsertString("project", r.Project.Name)
+	resourceAttrs.InsertString("cluster", r.Cluster.Name)
+	resourceAttrs.InsertString("hostname", r.Hostname)
+
+	data, err := json.Marshal(e)
+	if err != nil {
+		logger.Warn("failed to marshal", zap.Error(err))
+	}
+
+	t, err := time.Parse(layout, e.Timestamp.Date)
+	if err != nil {
+		logger.Warn("Time failed to parse correctly", zap.Error(err))
+	}
+	lr.SetTimestamp(pcommon.NewTimestampFromTime(t))
+	lr.SetObservedTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+
+	// The Message field contains description about the event,
+	// which is best suited for the "Body" of the LogRecordSlice.
+	lr.Body().SetStringVal(e.Message)
```

Comment:
The spec isn't very clear about this, and I'd love to get the opinion of others on what exactly should be in the body; I know we've typically been using the original, "raw" message as the body, so I'm not sure if that's preferred here over just the message field. 

IMO it makes more sense for Body to have the raw record, and for message to be included as an attribute.

See (again, the spec isn't clear either way):
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-body

Reply:
Personally, from that link alone, it looks like it wants the raw input as the body, and I agree that message can be an attribute, but the thing that really makes me convinced is that Body should be the raw input is the 

```
Can be for example a human-readable string message (including multi-line) describing the event in a free form or it can be a structured data composed of arrays and maps of other values. 
```

So I think it should just be the raw data itself. can add a message attribute later, thanks for looking out!